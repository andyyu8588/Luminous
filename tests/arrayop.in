// For compiler/VM testing purpose

a = [1, 2, 3];
b = ["a", "b", "c"];
c = [true, false, null];

d = a + 4;
e = b + "d";
f = c + (true equals false);

print(d);
print(e);
print(f);

g = d - 1;
h = e - 2;
i = f - 3;

print(g);
print(h);
print(i);

j = g * 2;
k = h * 10;
l = i * 1;

print(j);
print(k);
print(l);

print(size(j));
print(size(k));
print(size(l));

