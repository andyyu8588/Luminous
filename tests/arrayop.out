[1, 2, 3, 4]
[a, b, c, d]
[true, false, null, false]
[1, 3, 4]
[a, b, d]
[true, false, null]
[1, 3, 4, 1, 3, 4]
[a, b, d, a, b, d, a, b, d, a, b, d, a, b, d, a, b, d, a, b, d, a, b, d, a, b, d, a, b, d]
[true, false, null]
6
30
3
